core/pkg/server/config.go

```diff
@@ -0,0 +1,208 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/segmentio/encoding/json"
+	"github.com/wandb/wandb/core/internal/corelib"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+// A RunConfig representation.
+//
+// This is a type alias for refactoring purposes; it should be new type
+// otherwise.
+type RunConfigDict = map[string]interface{}
+
+// The configuration of a run.
+//
+// This is usually used for hyperparameters and some run metadata like the
+// start time and the ML framework used. In a somewhat hacky way, it is also
+// used to store programmatic custom charts for the run.
+//
+// The server process builds this up incrementally throughout a run's lifetime.
+type RunConfig struct {
```

Comment:
nice! can we move the `serializeConfig` function to be on a method of the RunConfig receiver i think it makes more sense now that you refactor this logic out of sender